macros/edr/tests/on_run_end/handle_tests_results.sql

```diff
@@ -53,53 +53,103 @@
 {% endmacro %}
 
 {% macro merge_data_monitoring_metrics(database_name, schema_name, test_metrics_tables) %}
-    {%- if test_metrics_tables %}
-        {%- set test_tables_union_query = elementary.union_metrics_query(test_metrics_tables) -%}
-        {%- set target_relation = adapter.get_relation(database=database_name, schema=schema_name, identifier='data_monitoring_metrics') -%}
-        {% if not target_relation %}
-          {% do exceptions.raise_compiler_error("Couldn't find Elementary's models. Please run `dbt run -s elementary`.") %}
-        {% endif %}
-        {%- set temp_relation = elementary.make_temp_view_relation(target_relation) -%}
-        {%- if test_tables_union_query %}
-            {{ elementary.file_log('Running union query from test tables to ' ~ temp_relation.identifier) }}
-            {%- do run_query(dbt.create_table_as(True, temp_relation, test_tables_union_query)) %}
-            {% set dest_columns = adapter.get_columns_in_relation(target_relation) %}
-            {{ elementary.file_log('Merging ' ~ temp_relation.identifier ~ ' to ' ~ target_relation.database ~ '.' ~ target_relation.schema ~ '.' ~ target_relation.identifier) }}
-            {%- if target_relation and temp_relation and dest_columns %}
-                {% set merge_sql = elementary.merge_sql(target_relation, temp_relation, 'id', dest_columns) %}
-                {%- do run_query(merge_sql) %}
-                {%- do adapter.commit() -%}
-                {{ elementary.file_log('Finished merging') }}
-            {%- else %}
-                {{ elementary.file_log('Error: could not merge to table: ' ~ target_name) }}
-            {%- endif %}
-        {%- endif %}
-    {%- endif %}
+    {%- if not test_metrics_tables %}
+      {% do return(none) %}
+    {% endif %}
+
+    {%- set test_tables_union_query = elementary.union_metrics_query(test_metrics_tables) -%}
+    {% if not test_tables_union_query %}
+      {% do return(none) %}
+    {% endif %}
+
+    {%- set target_relation = adapter.get_relation(database=database_name, schema=schema_name, identifier='data_monitoring_metrics') -%}
+    {% if not target_relation %}
+      {% do exceptions.raise_compiler_error("Couldn't find Elementary's models. Please run `dbt run -s elementary`.") %}
+    {% endif %}
+
+    {%- set temp_relation = elementary.make_temp_view_relation(target_relation) -%}
+    {% set insert_query %}
+      INSERT INTO {{ target_relation }} (
```

Comment:
This means that there could be multiple rows with the same `id`.
It probably means that we need to add an additional value to the [`id` construction](https://github.com/elementary-data/dbt-data-reliability/blob/539119bebb21ca9a035a9436f467f29b07c6f44b/macros/edr/data_monitoring/monitors_query/table_monitoring_query.sql#L74-L75) such as the invocation ID.